### PR TITLE
change tanzu imagepullsecret to tanzu secret registry

### LINKED ISSUE #723: Update tanzu imagepullsecret command to tanzu secret registry
Update tanzu imagepullsecret command to tanzu secret registry create/delete/update/get/list

### LINKED ISSUE #736: CLI plugins should not print usage unless -h is used
**Bug description**

plugins should not be printing usage on various failures. the error itself should have enough info to act on the problem. only if user uses --help then usage should pop up. if anything usage makes it very hard to find what the actual error is.

```
...snip...
Please consider using 'tanzu package installed update' to update the installed package with correct settings

Error: timed out waiting for the condition  #<---------- error reported to the user
Usage:                                      #<---------- usage is printed
  tanzu package install INSTALLED_PACKAGE_NAME --package-name PACKAGE_NAME --version VERSION [flags]

Examples:

    # Install package contour with installed package name as 'contour-pkg' with specified version and without waiting for package reconciliation to complete
    tanzu package install contour-pkg --package-name contour.tanzu.vmware.com --namespace test-ns --version 1.15.1-tkg.1-vmware1 --wait=false
...snip...
```

**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [ ] Addons
- [x] CLI
- [ ] Docs
- [ ] Installation
- [ ] Plugin
- [ ] Security
- [ ] Test and Release
- [ ] User Experience

**Expected behavior**

always silence usage. let user use standard -h flags to see it.

(example root cmd configuration in kapp that affects all commands: https://github.com/vmware-tanzu/carvel-kapp/blob/develop/pkg/kapp/cmd/kapp.go#L60 that eliminates this problem)

**Steps to reproduce the bug**

**Version** (include the SHA if the version is not obvious)

probably affects various plugins (incl package pluging, imagepullsecret plugin)

**Environment where the bug was observed (cloud, OS, etc)**

**Relevant Debug Output (Logs, manifests, etc)**

### PR DESCRIPTION
**Pull request by maralavi**
**What this PR does / why we need it:**
- This PR is to change plugin name "tanzu imagepullsecret" to "tanzu secret registry" (Fixes https://github.com/vmware-tanzu/tanzu-framework/issues/723)
- Fixes https://github.com/vmware-tanzu/tanzu-framework/issues/736

**Describe testing done for PR:**
Manual testing in the cluster and existing unit tests.

**Does this PR introduce a user-facing change?:**
None

**Release note**:

```release-note
The CLI plugin "imagepullsecret" has been renamed to "secret". Please use "tanzu plugin delete imagepullsecret" to delete the obsolete plugin.

```

**New PR Checklist**

- [X] Ensure PR contains only public links or terms
- [X] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [X] Squash the commits in this branch before merge to preserve our git history
- [ ] If this PR is just an idea or POC, use a [Draft PR](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests#draft-pull-requests) instead of a full PR
- [X] Add appropriate [kind label](../docs/release/kind-labels.md) according to what type of issue is being addressed.

### CODE REVIEW

File: cmd/cli/plugin/secret/README.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+#### Update a registry secret
+
+   The "update" command updates the v1/Secret of type kubernetes.io/dockerconfigjson.
+   In case of specifying the --export-to-all-namespaces flag, the SecretExport resource will also get updated. Otherwise, there will be no changes in the SecretExport resource.
+
+##### Update a registry secret. There will be no changes in the associated SecretExport resource
+
+   ```sh
+   >>> tanzu imagepullsecret update test-secret --username test-user -n test-ns --password-env-var PASSENV
```
Comment (shivaani0505) [inline cmd/cli/plugin/secret/README.md:136]: Need to change imagepullsecret => secret registry
Same with output

File: cmd/cli/plugin/secret/README.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+   \ Updating image pull secret 'test-secret'...
+     Updated image pull secret 'test-secret' in namespace 'test-ns'
+   ```
+
+##### Update a registry secret with 'export-to-all-namespaces' flag being set
+
+   ```sh
+   >>> tanzu imagepullsecret update test-secret--username test-user -n test-ns --password-env-var PASSENV --export-to-all-namespaces=true
```
Comment (shivaani0505) [inline cmd/cli/plugin/secret/README.md:144]: Same as above.
Need to change imagepullsecret => secret registry
Same with output for the command

File: cmd/cli/plugin/secret/README.md [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+   \ Updating image pull secret 'test-secret'...
+     Updated image pull secret 'test-secret' in namespace 'test-ns'
+     Exported image pull secret 'test-secret' to all namespaces
+   ```
+
+##### Update a registry secret with 'export-to-all-namespaces' flag being clear. In this case, the associated SecretExport resource will get deleted
+
+   ```sh
+   >>> tanzu imagepullsecret update test-secret --username test-user -n test-ns --password-env-var PASSENV --export-to-all-namespaces=false
```
Comment (shivaani0505) [inline cmd/cli/plugin/secret/README.md:166]: Same as above

File: cmd/cli/plugin/secret/README.md [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+   Warning: By specifying --export-to-all-namespaces as true, given secret contents will be available to ALL users in ALL namespaces. Please ensure that included registry credentials allow only read-only access to the registry with minimal necessary scope.
+   Are you sure you want to proceed? [y/N]: y
+
+   \ Updating image pull secret 'test-secret'...
+     Updated image pull secret 'test-secret' in namespace 'test-ns'
+     Exported image pull secret 'test-secret' to all namespaces
+   ```
+
+   ```sh
+   >>> tanzu imagepullsecret update test-secret --username test-user -n test-ns --password-env-var PASSENV --export-to-all-namespaces
```
Comment (shivaani0505) [inline cmd/cli/plugin/secret/README.md:154]: same as above

**Review by shivaani0505 — APPROVED** [review]
LGTM

File: cmd/cli/plugin/secret/main.go [hunk 5]
```diff
@@ -16,14 +16,13 @@ import (
 )
 
 var descriptor = cliv1alpha1.PluginDescriptor{
-	Name:        "imagepullsecret",
-	Description: "Manage image pull secret operations. Image pull secrets enable the package and package repository consumers to authenticate to private registries.",
+	Name:        "secret",
+	Description: "Tanzu secret management",
```
Comment (navidshaikh) [inline cmd/cli/plugin/secret/main.go:20]: if it should hint about the rename in the description and any instructions for user to clean up the old plugin ?(same for README)

**Comment by navidshaikh** [comment]
> or if there is any documentation needed in addition to that?

@maralavi: [cli getting started docs](https://github.com/vmware-tanzu/tanzu-framework/blob/main/docs/cli/getting-started.md) need to be updated as well.

**Comment by vijaykatam** [comment]
@navidshaikh I did not see any references to imagepullsecret in getting started docs.

**Review by vijaykatam — APPROVED** [review]
LGTM

**Comment by navidshaikh** [comment]
> I did not see any references to imagepullsecret in getting started docs.

@vijaykatam : CLI getting started docs dont mention any specific plugins. Added #914 to explicitly mention about deprecation with `v0.8.0` release.

**Pull request merged**